app/src/test/java/com/x8bit/bitwarden/ui/tools/feature/generator/GeneratorViewModelTest.kt

```diff
@@ -856,6 +857,65 @@ class GeneratorViewModelTest : BaseViewModelTest() {
         }
     }
 
+    @Test
+    fun `LifecycleResumedAction should use storage options derived state over VM state`() {
+        val initialState = initialUsernameState.copy(
+            selectedType = GeneratorState.MainType.Username(
+                selectedType = GeneratorState.MainType.Username.UsernameType.PlusAddressedEmail(
+                    email = "currentEmail",
+                ),
+            ),
+        )
+        val viewModel = createViewModel(initialState)
+        fakeGeneratorRepository.saveUsernameGenerationOptions(
+            UsernameGenerationOptions(
+                type = UsernameGenerationOptions.UsernameType.RANDOM_WORD,
+            ),
+        )
+        val expectedState = initialState.copy(
+            selectedType = GeneratorState.MainType.Username(
+                selectedType = GeneratorState.MainType.Username.UsernameType.RandomWord(),
+            ),
+            generatedText = "randomWord",
+        )
+        viewModel.trySendAction(GeneratorAction.LifecycleResume)
+        assertEquals(
+            expectedState,
+            viewModel.stateFlow.value,
+        )
+    }
+
+    @Test
+    fun `No LifecycleResumedAction should use VM state options derived state over VM state`() {
+        val initialState = initialUsernameState.copy(
+            selectedType = GeneratorState.MainType.Username(
+                selectedType = GeneratorState.MainType.Username.UsernameType.PlusAddressedEmail(
+                    email = "currentEmail",
+                ),
+            ),
+        )
+        val viewModel = createViewModel(initialState)
+        fakeGeneratorRepository.saveUsernameGenerationOptions(
+            UsernameGenerationOptions(
+                type = UsernameGenerationOptions.UsernameType.RANDOM_WORD,
+            ),
+        )
+        val stateIfUsingTheDiskOption = initialState.copy(
+            selectedType = GeneratorState.MainType.Username(
+                selectedType = GeneratorState.MainType.Username.UsernameType.RandomWord(),
+            ),
+            generatedText = "randomWord",
+        )
+        assertNotEquals(
+            stateIfUsingTheDiskOption,
+            viewModel.stateFlow.value,
+        )
+        assertEquals(
+            initialState.copy(generatedText = "email+abcd1234@address.com"),
+            viewModel.stateFlow.value,
```

Comment:
How does this assertion work, you are asserting the state is different without changing anything?

Reply:
So the "loadOptions" gets called on the `init` block, its hard to make this test meaningful since we are using a fake vs mock on the generator repo. Basically what I am trying to show is if `loadOptions(false)` is called it will use the VM stateFlow value vs pulling from disk.

Reply:
basically with the PlusAddressedEmail, when we loadOptions it generates that email in fake but that happens cause the init block.

Reply:
This still feels like a race condition, right?